Client/Assets/Favicons.js

```diff
@@ -2,47 +2,47 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-(function() {
- // These integers should be kept in sync with the IconType raw-values
- var ICON = 0;
- var APPLE = 1;
- var APPLE_PRECOMPOSED = 2;
- var GUESS = 3;
 
- var Favicons = {
-  selectors: { "link[rel~='icon']": ICON,
-              "link[rel='apple-touch-icon']": APPLE,
-              "link[rel='apple-touch-icon-precomposed']": APPLE_PRECOMPOSED },
 
- getAll: function() {
-    var res = {}
-    var foundIcons = false
 
-    for (selector in this.selectors) {
-      var icons = document.querySelectorAll(selector)
-      if (icons.length) {
-        res = {}
-      }
-      for (var i = 0; i < icons.length; i++) {
-        var href = icons[i].href;
-        res[href] = this.selectors[selector];
-          foundIcons = true
+window._FireFox_Favicons = function() {
```

Comment:
Please use the `__firefox__` namespace that we use in other content scripts. The initialization pattern we've been using is this:

```
if (!window.__firefox__) {
  window.__firefox__ = {};
}

window.__firefox__.favicons = ...
```

This keeps everything namespaced to a single global object.